datarobot_provider/operators/model_registry.py

```diff
@@ -0,0 +1,92 @@
+from enum import Enum
+from typing import Any
+from typing import Dict
+from typing import Sequence
+
+import datarobot as dr
+from airflow.exceptions import AirflowException
+from airflow.utils.context import Context
+
+from datarobot_provider.hooks.datarobot import DataRobotHook
+from datarobot_provider.operators.base_datarobot_operator import BaseDatarobotOperator
+
+
+class ModelType(Enum):
+    LEADERBOARD = "leaderboard"
+    CUSTOM = "custom"
+    EXTERNAL = "external"
+
+
+class CreateRegisteredModelVersionOperator(BaseDatarobotOperator):
+    """
+    Dynamically creates a registered model version using one of three methods:
+    - Leaderboard Model
+    - Custom Model Version
+    - External Model
+
+    :param model_version_params: Dictionary with parameters for creating model version.
+    :param datarobot_conn_id: Airflow connection ID for DataRobot.
+    """
+
+    template_fields: Sequence[str] = ["model_version_params"]
+    template_fields_renderers: dict[str, str] = {}
+    template_ext: Sequence[str] = ()
+    ui_color = "#f4a460"
+
+    def __init__(
+        self,
+        *,
+        model_version_params: Dict[str, Any],
+        datarobot_conn_id: str = "datarobot_default",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.model_version_params = model_version_params
+        self.datarobot_conn_id = datarobot_conn_id
+
+    def execute(self, context: Context) -> str:
+        """Executes the operator to create a registered model version."""
+        DataRobotHook(datarobot_conn_id=self.datarobot_conn_id).run()
+        model_type = self.model_version_params.get("model_type")
+
+        if not model_type:
+            raise ValueError("'model_type' must be specified in model_version_params.")
+
+        try:
+            model_type_enum = ModelType(model_type)
+        except ValueError:
+            raise AirflowException(f"Invalid model_type: {model_type}") from None
```

Comment:
Could you move these lines into the `validate()` method now provided by the newer base class?

I think lines should work as is as long as you alter:
`if not model_type:`
to be:
`if not self.model_version_params.get("model_type"):`